src/sinks/util/unix.rs

```diff
@@ -83,49 +95,43 @@ impl UnixConnector {
     }
 }
 
-impl Into<UnixSink> for UnixConnector {
-    fn into(self) -> UnixSink {
-        UnixSink::new(self.path)
+impl From<UnixConnector> for UnixService {
+    fn from(connector: UnixConnector) -> UnixService {
+        UnixService { connector }
     }
 }
 
-impl Into<UnixService> for UnixConnector {
-    fn into(self) -> UnixService {
-        UnixService { connector: self }
-    }
+pub struct UnixService {
+    connector: UnixConnector,
 }
 
-#[derive(Debug, Snafu)]
-pub enum UnixSocketError {
-    #[snafu(display("Connect error: {}", source))]
-    ConnectError { source: tokio::io::Error },
-    #[snafu(display("Send error: {}", source))]
-    SendError { source: tokio::io::Error },
-}
+impl tower::Service<Bytes> for UnixService {
+    type Response = ();
+    type Error = UnixStreamError;
+    type Future = BoxFuture<'static, Result<(), Self::Error>>;
 
-pub struct UnixSink {
-    connector: UnixConnector,
-    state: UnixSinkState,
-    backoff: ExponentialBackoff,
-}
+    fn poll_ready(&mut self, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
 
-type UnixSocket = FramedWrite<UnixStream, BytesCodec>;
-type UnixSocket01 = CompatSink<UnixSocket, Bytes>;
+    fn call(&mut self, msg: Bytes) -> Self::Future {
+        let connect = self.connector.connect();
+        async move { connect.await?.send(msg).await.context(SendError) }.boxed()
```

Comment:
We should not be initiating a new connection on every event. (I think you just moved this).

Reply:
I only moved this code inside the file. Issue about this: https://github.com/timberio/vector/issues/4610